contracts/number/types/Fixed18.sol

```diff
@@ -120,6 +120,22 @@ library Fixed18Lib {
         return Fixed18.wrap(Fixed18.unwrap(a) * BASE / Fixed18.unwrap(b));
     }
 
+    /**
+     * @notice Divides signed fixed-decimal `a` by `b`
+     * @param a Signed fixed-decimal to divide
+     * @param b Signed fixed-decimal to divide by
+     * @param roundOutwards Whether to round away from zero
+     * @return Resulting divided signed fixed-decimal
+     */
+    function div(Fixed18 a, Fixed18 b, bool roundOutwards) internal pure returns (Fixed18) {
+        if (!roundOutwards) return div(a, b);
+        if (isZero(a) && !isZero(b)) return ZERO;
+
+        int256 numerator = Fixed18.unwrap(a) * BASE;
+        int256 hasRemainder = (numerator % Fixed18.unwrap(b) != 0) ? int256(1) : int256(0);
+        return Fixed18.wrap(numerator / Fixed18.unwrap(b) + hasRemainder * sign(a) * sign(b));
```

Comment:
can we simplify this section to something like:

`sign(a.mul(b)).mul(a.abs().divCeil(b.abs()))`

I'm not sure if this would add a ton of gas though?

Reply:
Couple of comments here:
- We have to add in a zero-denominator check since OpenZeppelin's `ceilDiv` doesn't revert if `a` and `b` are both 0.
- `sign(a) * sign(b)` is a little safer in case `a.mul(b) overflows`. But the latter saves 41 gas (maybe because `sign` is expensive, haven't looked into it yet). @kbrizzle will defer to you on which one we go with here.
- Surprisingly, even the former saves 65 gas! (23025 vs 22960 gas with viaIR turned on). So I'll switch to something like:
```solidity
function divOut(Fixed18 a, Fixed18 b) internal pure returns (Fixed18) {
    if (Fixed18.unwrap(b) == 0) {
        revert Fixed18DivideByZeroError();
    }

    uint256 unsignedQuotient = Math.ceilDiv(UFixed18.unwrap(a.abs()) * uint256(BASE), UFixed18.unwrap((b.abs())));
    return Fixed18.wrap(sign(a) * sign(b) * (int256)(unsignedQuotient));
}
```

Reply:
I see yeah the naive `a * b` restricts the range more than it needs to. How about something like this, I think this is super clean. Basically trying to minimize the number of place we have custom math under the hood, and reuse other stuff as long as it doesn't cause tons of extra gas.

```
function divOut(Fixed18 a, Fixed18 b) internal pure returns (Fixed18) {
    if (Fixed18.unwrap(b) == 0) revert Fixed18DivideByZeroError();

    return Fixed18.from(sign(a) * sign(b), a.abs().divOut(b.abs()));
}
```

Reply:
Nice, using `from` is an awesome idea!

Reply:
Since this uses `UFixed18`'s `divOut` which does divide-by-zero reverts correctly, we also won't need the zero-denominator check :)

Reply:
yeah exactly!